merge_duplicate_contacts/wizard/merge_contact.py

```diff
@@ -151,71 +151,49 @@ def _action_new_next_screen(self):
-            "'ewe.net','ewetel.net','fastmail.com','fastmail.de','fastmail.fm','fastmail.net',"
-            "'free.fr','freenet.de','gmail.com','gmx.at','gmx.ch','gmx.com','gmx.de','gmx.eu',"
-            "'gmx.fr','gmx.info','gmx.li','gmx.org','gmxpro.de','gmx-topmail.de',"
-            "'googlemail.com','hotmail.ch','hotmail.co.uk','hotmail.com','hotmail.de',"
-            "'hotmail.es','hotmail.fr','hotmail.it','hush.com','hushmail.com','icloud.com',"
-            "'jpberlin.de','kabelmail.de','laposte.net','lavabit.com','librem.one',"
-            "'live.co.uk','live.com','live.de','live.fr','live.nl','mac.com','mail.de',"
-            "'mailbox.org','mailfence.com','me.com','meineinkauf.ch','msn.com',"
-            "'mykolab.ch','mykolab.com','netcologne.de','online.de','onlinehome.de',"
-            "'orange.fr','outlook.com','outlook.de','outlook.es','outlook.fr',"
-            "'posteo.at','posteo.ch','posteo.co.uk','posteo.de','posteo.eu',"
-            "'posteo.lu','posteo.net','posteo.org','proton.me','protommail.com',"
-            "'protonmail.ch','pt.lu','riseup.net','runbox.com','secure.mailbox.org',"
-            "'startmail.com','system.li','temp.mailbox.org','t-online.de',"
-            "'tuta.io','tutamail.com','tutanota.com','tutanota.de','vodafone.de',"
-            "'vodafonemail.de','wanadoo.fr','web.de','xs4all.nl','yahoo.ca','yahoo.co.uk',"
-            "'yahoo.com','yahoo.de','yahoo.es','yahoo.fr','yahoo.it')"
+        email_domains_config_value = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("merge_duplicate_contacts.email_domains")
         )
 
-        # Set up parts of the querries based on selected fields here
+        if email_domains_config_value:
+            email_domains = email_domains_config_value.split(",")
+        else:
+            email_domains = []
+
+        # Set up parts of the queries based on selected fields here
         # Match case would look better here but isn't a feature in our python version yet
+        sql_fields = []
+        where_queries = []
         for field in fields:
-            if field == "email":
-                group_by_querries.append("email")
-
-            if field == "name":
-                group_by_querries.append("name")
+            if field in ["email", "name"]:
+                sql_fields.append("lower(%s)" % field)
 
-            if field == "phone":
-                group_by_querries.append("phone")
+            elif field in ["vat"]:
+                sql_fields.append("replace(%s, ' ', '')" % field)
 
-            if field == "mobile":
-                group_by_querries.append("mobile")
+            else:
+                sql_fields.append(field)
 
-            if field == "is_company":
-                group_by_querries.append("is_company")
+        if email_domains and self.filter_domain_email:
+            email_domains = list(filter(None, (x.strip() for x in email_domains)))
```

Comment:
I actually meant all of it from get_param to `where_queries.append` inside of the `if self.filter_domain_email:`